src/components/LiquidityMining/Create/Steps/RewardAmount/index.tsx

```diff
@@ -1,175 +1,139 @@
-`
-
-const RewardInputLogo = styled(CurrencyLogo)`
-  position: absolute;
-  top: 8px;
-  right: 16px;
-`
+import { Currency, currencyEquals, TokenAmount } from '@swapr/sdk'
+import React, { useCallback, useMemo, useState } from 'react'
+import { Flex } from 'rebass'
 
-const StakablePairInputLogoContainer = styled.div`
-  position: absolute;
-  top: 8px;
-  right: 16px;
-`
+import { tryParseAmount } from '../../../../../state/swap/hooks'
 
-const FlexContainer = styled(Flex)`
-  ${props => props.theme.mediaWidth.upToExtraSmall`
-    flex-direction: column;
-  `}
-`
+import AssetSelector from '../PairAndReward/AssetSelector'
+import {
+  Actions,
+  ActionType,
+  CampaignType,
+  numberOfRewards,
+  RewardsObject,
+} from '../../../../../pages/LiquidityMining/Create'
+import CurrencySearchModal from '../../../../SearchModal/CurrencySearchModal'
+import styled from 'styled-components'
+import { ApprovalState } from '../../../../../hooks/useApproveCallback'
 
-const PoolSizeContainer = styled(Box)`
-  ${props => props.theme.mediaWidth.upToExtraSmall`
-    margin-top: 16px !important;
+const FlexWrapper = styled(Flex)`
+  gap: 28px;
+  margin-top: 32px !important;
+  width: fit-content;
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    gap:40px;
   `}
 `
-
 interface RewardAmountProps {
-  reward: TokenAmount | null
-  unlimitedPool: boolean
-  stakablePair: Pair | null
-  onRewardAmountChange: (newAmount: TokenAmount) => void
-  onUnlimitedPoolChange: (newValue: boolean) => void
-  onStakingCapChange: (newValue: TokenAmount | null) => void
+  rewardsObject: RewardsObject
+  setRewardsObject: React.Dispatch<Actions>
 }
 
-export default function RewardAmount({
-  reward,
-  unlimitedPool,
-  stakablePair,
-  onRewardAmountChange,
-  onUnlimitedPoolChange,
-  onStakingCapChange,
-}: RewardAmountProps) {
-  const previousReward = usePrevious(reward)
-  const [amount, setAmount] = useState('')
-  const [stakingCapString, setStakingCapString] = useState('')
+export default function RewardsSelection({ rewardsObject, setRewardsObject }: RewardAmountProps) {
+  const [currencySearchOpen, setCurrencySearchOpen] = useState<boolean>(false)
 
-  useEffect(() => {
-    if (!reward) setAmount('')
-  }, [reward])
+  const [currentReward, setCurrentReward] = useState<number | undefined>(undefined)
 
-  useEffect(() => {
-    if (unlimitedPool) {
-      setStakingCapString('')
-      onStakingCapChange(null)
-    }
-  }, [onStakingCapChange, stakablePair, unlimitedPool])
+  const handleDismissCurrencySearch = useCallback(() => {
+    setCurrencySearchOpen(false)
+  }, [])
+  const handelOpenPairOrTokenSearch = useCallback(value => {
+    setCurrentReward(value)
+    setCurrencySearchOpen(true)
+  }, [])
+  const disabledRewardsArray = useMemo(() => {
+    const mappedArray = rewardsObject.rewards.map(reward => reward?.currency)
+    const filteredArray = mappedArray.filter(currency => currency && currency !== undefined) as Currency[]
```

Comment:
Looking at this we can take advantage of functional programming and do something like this: 
```suggestion
    const filteredRewardsArray = rewardsObject.rewards.map(reward => reward?.currency).filter(currency => currency && currency !== undefined) as Currency[]
```